app/javascript/components/App.jsx

```diff
@@ -10,22 +13,79 @@ const App = () => {
         throw new Error('Network response was not ok.');
       }
       const data = await response.json();
+      console.log(data);
       setRounds(data);
     } catch (error) {
       console.log('An error occurred while fetching the data.');
     }
   }
 
+  // Filter data based on selected year and name
+  const filterDataByYearAndName = (year, name) => {
+    setSelectedYear(year);
+    setSelectedName(name);
+  }
+
+  // Filtered data based on selected year and name
+  //const filteredData = selectedYear ? rounds.filter(item => new Date(item.draw_at).getFullYear() === selectedYear) : rounds;
+  const filteredData = rounds.filter(
+    item => {
+      const yearMatch = !selectedYear || new Date(item.draw_at).getFullYear() === selectedYear;
+      const nameMatch = !selectedName || item.name === selectedName;
+      return yearMatch && nameMatch;
+    }
+  );
```

Comment:
這裡是一個小建議，可以分段篩，這樣未來要加新的條件會更方便
```suggestion
const filteredData = rounds.filter(item => {
  return !selectedYear || new Date(item.draw_at).getFullYear() === selectedYear 
}).filter(item => {
  return !selectedName || item.name.startsWith(selectedName);
});
```

Reply:
好的